doc/developers/contributing.rst

```diff
@@ -207,6 +207,154 @@ and Cython optimizations.
    <http://astropy.readthedocs.org/en/latest/development/workflow/development_workflow.html>`_
    sections.
 
+Rebasing and squashing commits
+------------------------------
+
+Resolving merge conflicts and Rebasing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you changed the same part of the same file differently in the two branches you’re merging
+together, Git won’t be able to merge them cleanly. This is called a merge conflict. To resolve a
+merge conflict you can either:
+	
+	* Open the files manually and resolve them as git adds standard resolution markers to the
+	  files that have conflicts.
+
+	* Use a graphical tool by running ``git mergetool``. This tool will walk you through all the
+	  conflicts.
+
+.. note::
+	Detailed guide on solving merge conflicts can be found on `git-scm.com
+	<http://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging#Basic-Merge-Conflicts>`_
+
+You should generally use ``rebase`` over ``merge`` to integrate changes from one branch into
+another branch. Merging creates a "merge commit" every time you incorporate upstream changes.
+This pollutes the feature branch's history and makes it hard for other developers to understand
+the history of the project. Whereas ``rebase`` effectively rewrites the project history by creating
+brand new commits for each commit in the original branch. This creates a much cleaner project history.
+To rebase:
+
+	1. Switch to feature branch::
+
+		$ git checkout my-feature-branch
+
+	2. Rebase on master::
+
+		$ git rebase master
+
+This can create merge conflicts and you might need to resolve them. If successfully rebased
+you can delete the backup branch by::
+
+	$ git branch -D my-feature-branch-bak
```

Comment:
Not sure if I commented this before (I remember thinking of doing so in my mind... Not sure if I actually did)

Do we need this backup branch thing?? It makes it seem a bit more complex than it is... We don't want people to get discouraged by a false assumption of complexity... We have git reflog I feel :)

You could maybe add a link saying, if you screw up there is git reflog. Refer this link on how to use reflog. (in a formal tone ofcourse ;))